lib/persisted-model.js

```diff
@@ -671,6 +697,28 @@ module.exports = function(registry) {
       rest: { after: convertNullToNotFoundError },
     });
 
+    var replaceByIdOptions = {
+      description: 'Replace attributes for a model instance and persist it into the data source.',
+      accessType: 'WRITE',
+      accepts: [
+        { arg: 'id', type: 'any', description: 'Model id', required: true,
+          http: { source: 'path' }},
+          { arg: 'data', type: 'object', http: { source: 'body' }, description:
+        'Model instance data' },
+      ],
+      returns: { arg: 'data', type: typeName, root: true },
+      http: { verb: 'post', path: '/:id/replace' },
+      rest: { after: convertNullToNotFoundError },
```

Comment:
IMO, `replaceById` should return 404 error when the record is not found, and therefore this `rest.after` hook should not be needed at all.